Source/WebCore/rendering/AutoTableLayout.cpp

```diff
@@ -52,17 +52,16 @@ void AutoTableLayout::recalcColumn(unsigned effCol)
     RenderTableCell* maxContributor = nullptr;
 
     for (auto& child : childrenOfType<RenderObject>(*m_table)) {
-        if (is<RenderTableCol>(child)) {
+        if (CheckedPtr column = dynamicDowncast<RenderTableCol>(child)) {
             // RenderTableCols don't have the concept of preferred logical width, but we need to clear their dirty bits
             // so that if we call setPreferredWidthsDirty(true) on a col or one of its descendants, we'll mark it's
             // ancestors as dirty.
-            downcast<RenderTableCol>(child).clearPreferredLogicalWidthsDirtyBits();
-        } else if (is<RenderTableSection>(child)) {
-            auto& section = downcast<RenderTableSection>(child);
-            unsigned numRows = section.numRows();
+            column->clearPreferredLogicalWidthsDirtyBits();
+        } else if (CheckedPtr section = dynamicDowncast<RenderTableSection>(child)) {
+            unsigned numRows = section->numRows();
             for (unsigned i = 0; i < numRows; ++i) {
-                RenderTableSection::CellStruct current = section.cellAt(i, effCol);
-                RenderTableCell* cell = current.primaryCell();
+                auto current = section->cellAt(i, effCol);
+                auto* cell = current.primaryCell();
```

Comment:
Yes, seems like it would be safer to use a CheckedPtr indeed. I didn't consistently adopt smart pointers in this patch since my aim was dynamicDowncast<>() adoption. I'll follow-up to adopt more smart pointers though (we have to be careful with those as they can impact perf and introduce new crashes).

Reply:
we are _inside_ layout and the render tree is supposed to stay intact for the duration of layout (we've got some release_asserts on this afair).